app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/reliefdetails/ReliefDetailsController.scala

```diff
@@ -381,6 +369,54 @@ class ReliefDetailsController @Inject() (
     }
   }
 
+  private def hasRequiredPreviousAnswerForOtherReliefs(answers: ReliefDetailsAnswers): Boolean =
+    answers match {
+      case IncompleteReliefDetailsAnswers(None, _, _)                                                  => false
+      case IncompleteReliefDetailsAnswers(Some(AmountInPence(0)), _, _)                                => true
+      case IncompleteReliefDetailsAnswers(Some(AmountInPence(_)), lettingsRelief, _)                   => lettingsRelief.isDefined
+      case CompleteReliefDetailsAnswers(AmountInPence(0), lettingsRelief, _) if !lettingsRelief.isZero => false
+      case CompleteReliefDetailsAnswers(AmountInPence(_), _, _)                                        => true
+    }
+
+  private def getRedirectForOtherReliefs(answers: ReliefDetailsAnswers): Call =
+    answers match {
+      case IncompleteReliefDetailsAnswers(None, _, _) => routes.ReliefDetailsController.privateResidentsRelief()
+      case IncompleteReliefDetailsAnswers(Some(AmountInPence(0)), _, _) =>
+        routes.ReliefDetailsController.privateResidentsRelief()
+      case IncompleteReliefDetailsAnswers(Some(AmountInPence(_)), lettingsRelief, _) =>
+        routes.ReliefDetailsController.lettingsRelief()
+      case CompleteReliefDetailsAnswers(AmountInPence(0), lettingsRelief, _) if !lettingsRelief.isZero =>
+        routes.ReliefDetailsController.privateResidentsRelief()
+      case CompleteReliefDetailsAnswers(AmountInPence(_), _, _) => routes.ReliefDetailsController.lettingsRelief()
+    }
+
+  private def hasRequiredPreviousAnswerForLettingsReliefs(answers: ReliefDetailsAnswers): Boolean =
+    answers match {
```

Comment:
you probably don't need all these cases. The previous question to letting relief is always private residents relief, so this becomes simply `answers.fold(_.privateResidentsRelief, c => Some(c.privateResidentsRelief)).isDefined`

Reply:
Done.